client/my-sites/checklist/main.jsx

```diff
@@ -28,6 +28,39 @@ import { isJetpackSite, getSiteSlug } from 'state/sites/selectors';
 import { recordTracksEvent } from 'state/analytics/actions';
 
 class ChecklistMain extends PureComponent {
+	componentDidMount() {
+		this.maybeRedirectJetpack();
+	}
+
+	componentDidUpdate( prevProps ) {
+		this.maybeRedirectJetpack( prevProps );
+	}
+
+	/**
+	 * Redirect Jetpack checklists to /plans/my-plan/:siteSlug
+	 *
+	 * Guard for relevant props updated and correct conditions
+	 *
+	 * @param {Object}   prevProps           Optional. Previous props from componentDidUpdate.
+	 * @param {?boolean} prevProps.isAtomic  Previous isAtomic
+	 * @param {?boolean} prevProps.isJetpack Previous isJetpack
+	 * @param {?string}  prevProps.siteSlug  Previous siteSlug
+	 */
+	maybeRedirectJetpack( { isAtomic, isJetpack, siteSlug } = {} ) {
+		if (
+			this.props.siteSlug &&
+			false === this.props.isAtomic &&
+			this.props.isJetpack &&
+			some( [
+				this.props.siteSlug !== siteSlug,
+				this.props.isAtomic !== isAtomic,
+				this.props.isJetpack !== isJetpack,
+			] )
+		) {
```

Comment:
Semantic nitpick, I think it'd be clearer if you kept the `prevProps` arg un-destructured, and destructured `this.props` instead (since a criterion like `siteSlug !== prevProps.siteSlug` is less confusing IMO -- here, the `prev` aspect is explicit).

Reply:
Agreed 👍 
df1274eaea1abf0b745d161f8dc9995042e9af13